src/wp-includes/kses.php

```diff
@@ -981,7 +982,18 @@ function wp_kses_split( $content, $allowed_html, $allowed_protocols ) {
 	$pass_allowed_html      = $allowed_html;
 	$pass_allowed_protocols = $allowed_protocols;
 
-	return preg_replace_callback( '%(<!--.*?(-->|$))|(<[^>]*(>|$)|>)%', '_wp_kses_split_callback', $content );
+	$token_pattern = <<<REGEX
+~
+	(                         # Detect comments of various flavors before attempting to find tags.
+		(?:<!--.*?(--!?>|$))  # Well-formed HTML comments like `<!-- ... -->` and also with invalid `--!>` closer.
+		|
+		</[^a-zA-Z][^>]*>     # Closing tags with invalid tag names.
```

Comment:
That regexp used to match everything between < and > but now it only matches invalid tag closers. So this change means valid tag names and valid closers are no longer treated by `_wp_kses_split_callback`. At the same time, the goal of this PR is to preserve HTML funky comments. Why, in order to preserve funky comments, is this passing _less_ information to the replacement callback?

Reply:
Aha – I'm wrong. This entire regexp branch is _a new clause_. Tag-like spans are still matched in the next branch. Cool.